### PR TITLE
Bug 1828719: Retry updating config resource during tests.

### PR DESCRIPTION
**Pull request by ricardomaraschini**
Tests were flaking on this as the operator is updating the resource
while the test runs.

Error being reported during tests:

```
managementstate_test.go:149: Operation cannot be fulfilled on configs.imageregistry.operator.openshift.io "cluster": the object has been modified; please apply your changes to the latest version and try again
```

### CODE REVIEW

File: test/e2e/managementstate_test.go [hunk 1]
```diff
@@ -141,11 +141,27 @@ func TestRemovedToManagedTransition(t *testing.T) {
 	}
 
 	t.Log("updating ManagementState to Managed with no storage config")
-	cr.Spec.ManagementState = operatorapi.Managed
-	cr.Spec.Storage = imageregistryv1.ImageRegistryConfigStorage{}
-	if _, err = te.Client().Configs().Update(
-		context.Background(), cr, metav1.UpdateOptions{},
-	); err != nil {
+	err = wait.Poll(time.Second, framework.AsyncOperationTimeout, func() (stop bool, err error) {
```
Comment (dmage) [inline test/e2e/managementstate_test.go:144]: It might be better to use `retry.RetryOnConflict`

Comment (ricardomaraschini) [reply test/e2e/managementstate_test.go]: Way better. Done.

**Comment by adambkaplan** [comment]
@ricardomaraschini now that we are past feature freeze, please create a BZ for this

**Comment by ricardomaraschini** [comment]
> @ricardomaraschini now that we are past feature freeze, please create a BZ for this

Done.

**Pull request merged**